### ISSUE TITLE
Assorted format inconsistencies

### ISSUE THREAD

**Issue by joshop** [BleuBleu/FamiStudio#105]
I've noticed a few pretty minor inconsistencies to do with file formats. In this case, the FTM for [this Cave Story VRC6 arrangement ](http://forums.famitracker.com/viewtopic.php?t=441):
- With a lot of the songs (easily noticeable on song 32, for instance), the loop point is one pattern too early. I could see this being because FamiTracker uses zero indexing and FamiStudio uses one indexing for pattern numbers.
- Taking song 33 and exporting to NSF causes some really strange behavior with square channel 1 that doesn't occur when just listening to it. This appears to be because FamiStudio _saturates_ notes going below the minimum pitch while playing, but _wraps around_ when exporting, causing the beginning of each note (which is lower due to an arpeggio envelope) to underflow and instead become extremely high.

**Comment by BleuBleu** [comment]
Hi. First of all. Id like you to read the documentation regarding FTM compatibility. If something is mentionned in there, its not a bug and I wont investigate it. 

Once you have read it, please amend the report and I will focus on the read issues.

https://famistudio.org/doc/import/#famitracker-text-or-binary

-Mat

**Comment by joshop** [comment]
Sorry about that, I hadn't seen that page. I don't think the other two are covered there, though I might have missed it.

**Comment by BleuBleu** [comment]
Awesome! 

Unfortunately, ill give you the same answer as last time : I wont be home until jan 30th where i have access to all my hardware. Ill get on these immediately when i get back!

-Mat

**Comment by BleuBleu** [comment]
I just released 3.2.3 and realized i *completely* forgot to investigate your 2 issues. Many apologies for this. I will get to them very soon.

**Comment by BleuBleu** [comment]
Hi. Sorry again for the delay. Both of these will be addressed in 3.3.0, which should come out early-may.

1) There indeed was an issue with songs where the jump wasn't at the very end of the song. FamiStudio will not truncate the song and give a warning when that happens. This will mean that the "unreachable" patterns will not be imported. 
```
Warning: Patterns at end of song 'And Then Everyone Died...' are unreachable due to a previous jump (Bxx), truncating.
```

2) You are spot on again. Since we want to keep the NSF lightweight and fast, we dont want to do clamping in there. So what i suggest is adding an option in FamiStudio do disable the clamping. This will help debug these kind of situations (see image).

![image](https://user-images.githubusercontent.com/6654334/163333598-7fc34738-85a6-4b7d-9ccc-4a4f501d0ce8.png)

-Mat

**Comment by BleuBleu** [comment]
Fixed 3.3.0! Closing.
